### ISSUE TITLE
v3.helm.Release reports incorrect kube version

### ISSUE THREAD

**Issue by awopenphone** [pulumi/pulumi-kubernetes#2752]
### What happened?

cert-manager chart from https://charts.jetstack.io

### Example

      kubernetes:helm.sh/v3:Release cert-manager  error: kubernetes:helm.sh/v3:Release resource 'cert-manager': property chart value {cert-manager} has a problem: chart requires kubeVersion: >= 1.22.0-0 which is incompatible with Kubernetes v1.20.0; check the chart name and repository configuration.

### Output of `pulumi about`

❯ pulumi about       
CLI          
Version      3.101.1
Go Version   go1.21.6
Go Compiler  gc

Plugins
NAME        VERSION
aws         6.18.0
aws         6.18.0
aws         6.18.0
aws-native  0.79.0
awsx        2.4.0
awsx        2.4.0
awsx        2.4.0
docker      4.5.0
docker      4.5.0
docker      4.5.0
docker      3.6.1
docker      3.6.1
docker      3.6.1
kubernetes  4.5.6
kubernetes  4.5.6
nodejs      unknown

Host     
OS       darwin
Version  14.2.1
Arch     arm64

This project is written in nodejs: executable='/Users/andrew/.nvm/versions/node/v18.18.2/bin/node' version='v18.18.2'

Current Stack: snip

Found no pending operations associated with snip

Backend        
Name           pulumi.com
snip
Token type     personal

Dependencies:
NAME                               VERSION
@pulumi/pulumi                     3.101.1
@swc/jest                          0.2.29
@types/node                        18.19.7
jest                               29.7.0
@pulumi/aws                        6.18.0
@pulumi/compliance-policy-manager  0.1.2
@pulumi/kubernetes                 4.5.6
@pulumi/policy                     1.9.0
@types/jest                        29.5.11
@pulumi/aws-compliance-policies    0.1.0
@pulumi/awsx                       2.4.0

### Additional context

This issue was present before November 2023 and was fixed but I got this again on a Github Action run with all of the latest dependencies. 

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by awopenphone** [comment]
this chart does it too 

```
kubernetes:helm.sh/v3:Release elastic-operator  error: kubernetes:helm.sh/v3:Release resource 'elastic-operator': property chart value {eck-operator} has a problem: chart requires kubeVersion: >=1.21.0-0 which is incompatible with Kubernetes v1.20.0; check the chart name and repository configuration.
```

**Comment by mjeffryes** [comment]
I think this is likely fixed in https://github.com/pulumi/pulumi-kubernetes/issues/2739 just waiting on a release. I will see about getting a new release out as soon as possible.

**Comment by EronWright** [comment]
This issue was mainly fixed in @pulumi/kubernetes v4.6.0 (see https://github.com/pulumi/pulumi-kubernetes/pull/2672).  Sorry for the inconvenience, please update and try again.

**Comment by awopenphone** [comment]
This issue is resolved my libs were out of date sorry.
